### PR TITLE
Added ensure-static-ips.sh script

### PR DESCRIPTION
**Pull request by bartsmykla**
/cc @dims 

ref. https://github.com/kubernetes/k8s.io/issues/775

Signed-off-by: Bart Smykla <bsmykla@vmware.com>

### CODE REVIEW

File: infra/gcp/ensure-static-ips.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    echo "usage: $0" > /dev/stderr
+    echo > /dev/stderr
+}
+
+if [ $# != 0 ]; then
+    usage
+    exit 1
+fi
+
+function create_address() {
```
Comment (spiffxp) [inline infra/gcp/ensure-static-ips.sh:39]: nit: naming
```suggestion
function ensure_address() {
```

File: infra/gcp/ensure-static-ips.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    "gcsweb-k8s-io"
+    "k8s-io-ingress-canary"
+    "k8s-io-ingress-prod"
+    "perf-dash-k8s-io-ingress-prod"
+    "slack-infra-ingress-prod"
+)
+
+for address in "${ADDRESSES[@]}"; do
+    color 6 "Create address: $address"
+    create_address "$address"
```
Comment (spiffxp) [inline infra/gcp/ensure-static-ips.sh:67]:
```suggestion
    ensure_address "$address"
```

File: infra/gcp/ensure-static-ips.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return 1
+    fi
+    local address_name="$1"
+
+    # Create address if it doesn't exist
+    if ! gcloud compute addresses describe "$address_name" --global \
+        --project "$PROJECT_NAME" > /dev/null 2>&1; then
+        gcloud compute addresses create "$address_name" --global \
+            --project "$PROJECT_NAME"
+        echo "Address: $address_name created successfully"
```
Comment (thockin) [inline infra/gcp/ensure-static-ips.sh:51]: Maybe we should print the value of it?

File: infra/gcp/ensure-static-ips.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+function create_address() {
+    if [ $# != 1 ]; then
+        echo "create_address(name) requires 1 argument" >&2
+        return 1
+    fi
+    local address_name="$1"
+
+    # Create address if it doesn't exist
+    if ! gcloud compute addresses describe "$address_name" --global \
```
Comment (thockin) [inline infra/gcp/ensure-static-ips.sh:47]: in the past we had some Service with external IPs (as opposed to Ingress), which requires a slightly different command (no --global, not sure if there's more).

I don't think we have any of those left, but maybe a comment here acknowledging it

File: infra/gcp/ensure-static-ips.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+fi
+
+function create_address() {
+    if [ $# != 1 ]; then
+        echo "create_address(name) requires 1 argument" >&2
+        return 1
+    fi
+    local address_name="$1"
+
+    # Create address if it doesn't exist
```
Comment (thockin) [inline infra/gcp/ensure-static-ips.sh:46]: I could see this moving to lib.sh as `ensure_global_ip(name)`

File: infra/gcp/ensure-static-ips.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    if ! gcloud compute addresses describe "$address_name" --global \
+        --project "$PROJECT_NAME" > /dev/null 2>&1; then
+        gcloud compute addresses create "$address_name" --global \
+            --project "$PROJECT_NAME"
+        echo "Address: $address_name created successfully"
+    else
+        echo "Address: $address_name already exists"
```
Comment (thockin) [inline infra/gcp/ensure-static-ips.sh:53]: we don't usually print on success

File: infra/gcp/ensure-static-ips.sh (hunk 3 above)
Comment (bartsmykla) [reply infra/gcp/ensure-static-ips.sh:51]: We actually can't get it easily, as it looks like it's async operation and in response we are receiving the url address only:
```
Created [https://www.googleapis.com/compute/v1/projects/kubernetes-public/global/addresses/foo].
```

File: infra/gcp/ensure-static-ips.sh (hunk 4 above)
Comment (bartsmykla) [reply infra/gcp/ensure-static-ips.sh]: I'm not sure what should I put in the comment now, when the logic was moved to separate function in `lib.sh` file

File: infra/gcp/ensure-static-ips.sh (hunk 2 above)
Comment (bartsmykla) [reply infra/gcp/ensure-static-ips.sh]: Moved the function to the `lib.sh` file, and addressed the name change

File: infra/gcp/ensure-static-ips.sh (hunk 1 above)
Comment (bartsmykla) [reply infra/gcp/ensure-static-ips.sh]: I moved the function to Moved the function to the `lib.sh` file as @thockin suggested, and addressed the name change

File: infra/gcp/ensure-static-ips.sh (hunk 6 above)
Comment (bartsmykla) [reply infra/gcp/ensure-static-ips.sh]: Addressed

File: infra/gcp/ensure-static-ips.sh (hunk 5 above)
Comment (bartsmykla) [reply infra/gcp/ensure-static-ips.sh]: Done

**Comment by spiffxp** [comment]
whoops, I ended up redoing some of this in https://github.com/kubernetes/k8s.io/pull/806

I'll merge this and rebase

**Pull request merged**